tasmota/tasmota_xdrv_driver/xdrv_02_9_mqtt.ino

```diff
@@ -737,6 +737,21 @@ void MqttPublishPayload(const char* topic, const char* payload) {
   MqttPublishPayload(topic, payload, 0, false);
 }
 
+void MqttPublish(const char* topic, bool retained, bool binary) {
+  // Publish <topic> default ResponseData string with optional retained
+  char* response_data;
+  uint32_t binary_length = 0;
+  response_data = ResponseData();
+
+  if (binary) {
+    if (strlen(response_data) > 3 && strncmp(response_data, "0x", 2) == 0) {
+      binary_length = HexStringToBinaryString(&response_data);
+    }
+  }
+
+  MqttPublishPayload(topic, response_data, binary_length, retained);
+}
+
 void MqttPublish(const char* topic, bool retained) {
```

Comment:
Is this one still needed?

Reply:
The function `void MqttPublish(const char* topic, bool retained)` is used in xdrv_12_home_assistant.ino, xdrv_27_shutter.ino and a few others.
The function could just look for strings in the format `0x([0-9A-Za-z]{2})+` but that might cause a breaking change where someone is trying to send a string and not binary data.